src/app/inventory/store/objectives.ts

```diff
@@ -61,3 +62,53 @@ export function isBooleanObjective(
       (!objectiveDef.allowOvercompletion || !objectiveDef.showValueOnComplete))
   );
 }
+
+export function isTrialsPassage(item: DimItem, defs: D2ManifestDefinitions) {
+  if (item.objectives?.length === 3 && item.isExotic) {
+    for (const objective of item.objectives) {
+      const objectiveDef = defs.Objective.get(objective.objectiveHash);
+      if (isFlawlessObjective(objective, objectiveDef)) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+export function isFlawlessPassage(
```

Comment:
I think this needs documentation - I didn't realize until reading the code that it meant "is the trials passage flawless?"

Reply:
I've added a comment to explain it.